src/activation.js

```diff
@@ -180,11 +180,58 @@ function shouldContinue(output, router: Router) {
   return output;
 }
 
+//wraps a subscription, allowing unsubscribe calls even if
+//the first value comes synchronously
+class SafeSubscription {
+  //if this were TypeScript, subscriptionFunc would be of type
+  //(sub: SafeSubscription) => Subscription
+  constructor(subscriptionFunc) {
+    this._subscribed = true;
+    this._subscription = subscriptionFunc(this);
+
+    if (!this._subscribed) this.unsubscribe();
+  }
+
+  get subscribed() {
+    return this._subscribed;
+  }
+
+  unsubscribe() {
+    if (this._subscribed && this._subscription) this._subscription.unsubscribe();
+
+    this._subscribed = false;
+  }
+}
+
 function processPotential(obj, resolve, reject) {
   if (obj && typeof obj.then === 'function') {
     return Promise.resolve(obj).then(resolve).catch(reject);
   }
 
+  if (obj && typeof obj.subscribe === 'function') {
+    //an object with a subscribe function should be assumed to be an observable
+    return new SafeSubscription(sub => obj.subscribe({
+      next() {
+        if (sub.subscribed) {
```

Comment:
you mean the check for whether the subscription is active? It's really there to avoid an extra resolve or reject (say, for a single value then complete), just kinda put the unsubscribe call in the block because... well, because.